### ISSUE TITLE
Package drop down only displays the first 100 or so packages

### ISSUE THREAD

**Issue by matt-lacross** [renevanosnabrugge/vsts-promotepackage-task#48]
We have a couple thousand packages and would love to use this but it not feasible because the package only displays 100 packages in the drop down.  Can you fix it so the drop down displays all of our packages from our VSTS feed?

**Comment by renevanosnabrugge** [comment]
Thanks for your comment.

Most of my users suffice by this functionality. If you want to support this , I happily accept your pull request!

Thanks
Rene

Verstuurd vanaf mijn iPhone

Op 2 mrt. 2020 om 13:37 heeft matt-lacross <notifications@github.com> het volgende geschreven:

﻿

We have a couple thousand packages and would love to use this but it not feasible because the package only displays 100 packages in the drop down. Can you fix it so the drop down displays all of our packages from our VSTS feed?

**Comment by jessehouwing** [comment]
The dropdown is editable. Just paste in the correct package name.

**Comment by armanik11** [comment]
@jessehouwing , @renevanosnabrugge 
Is **first 100** packages a limitation of the VSTS REST API Call made to retrieve and display packages? Is it specific to task UI? Because, when I call the same API (https://feeds.dev.azure.com/<Account>/_apis/packaging/Feeds/<FeedName>/packages?api-version=6.1-preview.1) via POSTMAN or Browser on a feed with thousands of packages, I see that it returns 1000 packages.

 "endpointUrl": "{{endpoint.url}}/{{ #regex ([a-fA-F0-9\\-]+/)[a-fA-F0-9\\-]+ feed }}_apis/packaging/feeds/{{ #regex (?:[a-fA-F0-9\\-]+/)?([a-fA-F0-9\\-]+) feed }}/**packages**?includeUrls=false",

**Comment by armanik11** [comment]
@jessehouwing , @renevanosnabrugge : Any pointers or ideas on my question above?

**Comment by armanik11** [comment]
@matt-lacross : Have you found any workarounds for this yet?

**Comment by jessehouwing** [comment]
Nope. The pointer is not to rely on the drop down and just type the desired value.

There may be a query option to do a type to search option, I haven't tried with that yet.

**Comment by armanik11** [comment]
Thanks @jessehouwing. I assume it won't fail to load the UI but rather just populate it with 100 packages correct?

**Comment by jessehouwing** [comment]
Yups. The API has a few interesting features I haven't been able to completely comprehend. Sometimes the API happily returns 100s of items and then the same call sometimes uses paging.

Unfortunately the task.json format doesn't understand the continuation token to continue fetching in the background or offer a auto-load feature as you scroll down.
